flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/calcite/Sink.scala

```diff
@@ -42,7 +42,8 @@ import scala.collection.JavaConversions._
  * @param hints
  *   the hints
  * @param targetColumns
- *   the specified target columns.
+ *   the specified target columns. @Deprecated(since = "2.2"), use
```

Comment:
```
the specified target columns. This field has been deprecated as part of FLIP-506 since 2.2. Please use
 *   [[TargetColumnWritingSpec]] in [[abilitySpecs]] instead.
```

Additionally, we should move all `abilitySpecs` from all the `Sink` subclasses up into the `Sink` base class. When a feature on the `Sink` is deprecated, we can provide a direct, compatible replacement for it.